src/getIdentifier.ts

```diff
@@ -0,0 +1,81 @@
+import { VerificationFragment } from "./types/core";
+import { Identity } from "./verifiers/issuerIdentity/dnsTxt";
+import { SignatureVerificationFragment } from "./verifiers/issuerIdentity/did";
+import { DnsVerificationFragment } from "./verifiers/issuerIdentity/dnsDid";
+
+enum IdentityProof {
+  DNS = "OpenAttestationDnsTxtIdentityProof",
+  DNSDID = "OpenAttestationDnsDidIdentityProof",
+  DID = "OpenAttestationDidIdentityProof",
+}
+
+const getDnsIdentifierProof = (fragment: VerificationFragment) => {
+  const type = "DNS";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: Identity) => ({
+      identifier: issuer.location,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data.identifier,
+    type,
+  };
+};
+
+const getDnsDidIdentifierProof = (fragment: VerificationFragment) => {
+  const type = "DNS-DID";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: DnsVerificationFragment) => ({
+      identifier: issuer.location,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data.location,
+    type,
+  };
+};
+
+const getDidIdentifierProof = (fragment: VerificationFragment) => {
+  const type = "DID";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: SignatureVerificationFragment) => ({
+      identifier: issuer.did,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data.did,
+    type,
+  };
+};
+
+export const getIdentityProofFragment = (fragments: VerificationFragment[]) => {
+  if (fragments.length < 1) {
+    throw new Error("Please provide at least one verification fragment");
+  }
+  return fragments.find((status) => status.type === "ISSUER_IDENTITY" && status.status === "VALID");
+};
+
+export const getIdentifier = (fragment: ReturnType<typeof getIdentityProofFragment>) => {
```

Comment:
I didn't clone locally, but is the return type correct ? For instance we could expect type tp be either "DNS" | "DID" | "DNS-DID" | "unknown". I think it will be inferred as string (I might be wrong too)

Reply:
return type is `VerificationFragment<any> | undefined`, which will be further refined after the guards so to me its robust enough.